stage0/src/image_header.rs

```diff
@@ -56,6 +56,22 @@ impl Image {
         self.0 as *const ImageVectors as u32
     }
 
+    #[cfg(feature = "dice")]
+    fn get_img_size(&self) -> Option<usize> {
+        use core::convert::TryFrom;
+
+        usize::try_from((unsafe { &*self.get_header() }).total_image_len).ok()
+    }
+
+    #[cfg(feature = "dice")]
+    pub fn as_bytes(&self) -> &[u8] {
```

Comment:
This feels like a place where an attacker would try to confuse stage0 by tampering with the img_size.  If I set the img_size to 1, the Cdi will only include the first byte but we'll boot the image, which must be more than 1 byte long, anyway.  We probably need to hash the entire image slot, not just the used area so we aren't relying on attacker-provided data.

Reply:
I've fleshed out my understanding of the threat and mitigations in an issue & have added it to the Trust Quorum backlog: https://github.com/oxidecomputer/hubris/issues/736